src/workspaces/features/SidebarMenu/SidebarMenu.tsx

```diff
@@ -106,39 +146,62 @@ const SidebarMenu = (props: SidebarMenuProps) => {
           <section>
             <div className="flex w-full items-center justify-between px-4 py-2 text-sm font-medium tracking-wide text-gray-500 opacity-90">
               {t("Your workspaces")}
+
               <button
                 type="button"
+                onClick={() => setDialogOpen(true)}
                 title={t("Create a new workspace")}
                 className="text-gray-400 hover:text-gray-600"
               >
                 <PlusCircleIcon className="h-5 w-5 " />
               </button>
+              <CreateWorkspaceDialog
+                open={isDialogOpen}
+                onClose={() => setDialogOpen(false)}
+              />
             </div>
 
-            {WORKSPACES.map((item) => (
-              <Link
-                noStyle
-                href={{
-                  pathname: "/workspaces/[workspaceId]",
-                  query: { workspaceId: item.id },
-                }}
-                className="flex items-center py-2.5 px-4 hover:bg-gray-100"
-                key={item.id}
-              >
-                {item.country && (
-                  <div className="mr-2.5 flex h-full items-center">
-                    <img
-                      alt="Country flag"
-                      className="h-4 flex-shrink rounded-sm"
-                      src={`/static/flags/${item.country.code}.gif`}
-                    />
+            <div className="max-h-96 overflow-y-auto">
+              {data &&
+                data?.workspaces.items.map((workspace, index) => (
+                  <Link
+                    noStyle
+                    href={{
+                      pathname: "/workspaces/[workspaceId]",
+                      query: { workspaceId: workspace.id },
+                    }}
+                    className="flex items-center py-2.5 px-4 hover:bg-gray-100"
+                    key={index}
+                  >
+                    {workspace.countries &&
+                      workspace.countries.length === 1 && (
+                        <div className="mr-2.5 flex h-full items-center">
+                          <img
+                            alt="Country flag"
+                            className="h-4 flex-shrink rounded-sm"
+                            src={workspace.countries[0].flag}
+                          />
+                        </div>
+                      )}
+                    <span className="text-sm leading-tight tracking-tight">
+                      {workspace.name}
+                    </span>
+                  </Link>
+                ))}
+              {data &&
+                data?.workspaces.totalItems !==
+                  data.workspaces.items.length && (
+                  <div className="pb-2 text-center">
+                    <button
+                      onClick={() => showMore(data.workspaces.totalItems)}
```

Comment:
No need to pass an argument since `showMore` already has access to `data`

Reply:
Okay okay